crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -772,11 +773,27 @@ impl MoveTestAdapter<'_> for SuiTestAdapter {
                 let epoch = self.get_latest_epoch_id()?;
                 Ok(Some(format!("Epoch advanced: {epoch}")))
             }
-            SuiSubcommand::AdvanceClock(AdvanceClockCommand { duration_ns }) => {
-                let effects = self
-                    .executor
-                    .advance_clock(Duration::from_nanos(duration_ns))
-                    .await?;
+            SuiSubcommand::AdvanceClock(advance_clock_command) => {
+                let duration = match advance_clock_command {
+                    AdvanceClockCommand {
+                        duration: Some(_),
+                        duration_ns: Some(_),
+                    } => panic!("Must specify only one of `--duration` or `--duration_ns`"),
+                    AdvanceClockCommand {
+                        duration: Some(duration),
+                        duration_ns: None,
+                    } => duration.parse::<IsoDuration>().unwrap().into(),
+                    AdvanceClockCommand {
+                        duration: None,
+                        duration_ns: Some(duration_ns),
+                    } => Duration::from_nanos(duration_ns),
+                    AdvanceClockCommand {
+                        duration: None,
+                        duration_ns: None,
+                    } => panic!("Must specify either `--duration` or `--duration_ns`"),
+                };
```

Comment:
Note that this function returns `anyhow` errors, rather than panics on failure.


```suggestion
                    } => bail!("Must specify only one of `--duration` or `--duration_ns`"),
                    AdvanceClockCommand {
                        duration: Some(duration),
                        duration_ns: None,
                    } => duration.parse::<IsoDuration>()?.into(),
                    AdvanceClockCommand {
                        duration: None,
                        duration_ns: Some(duration_ns),
                    } => Duration::from_nanos(duration_ns),
                    AdvanceClockCommand {
                        duration: None,
                        duration_ns: None,
                    } => bail!("Must specify either `--duration` or `--duration_ns`"),
                };
```